neutone_sdk/benchmark.py

```diff
@@ -0,0 +1,126 @@
+import timeit
+import itertools
+import click
+import torch
+from neutone_sdk.utils import load_neutone_model
+import numpy as np
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option("--model_file", help="Path to model file")
+@click.option(
+    "--buffer_size",
+    default=(128, 256, 512, 1024, 2048),
+    multiple=True,
+    help="Buffer sizes to benchmark",
+)
+@click.option(
+    "--sample_rate",
+    default=(48000,),
+    multiple=True,
+    help="Sample rates to benchmark",
+)
+@click.option("--repeat", default=10, help="How many times to repeat the benchmark")
+@click.option(
+    "--number",
+    default=30,
+    help="How many times to run each benchmark for each repetition",
+)
+@click.option("--num_threads", default=1, help="num_threads to use for the benchmark")
+@click.option(
+    "--num_interop_threads",
+    default=1,
+    help="num_interop_threads to use for the benchmark",
+)
+def benchmark_speed(
+    model_file,
+    buffer_size,
+    sample_rate,
+    repeat,
+    number,
+    num_threads,
+    num_interop_threads,
+) -> None:
+    torch.set_num_threads(num_threads)
+    torch.set_num_interop_threads(num_interop_threads)
+    with torch.inference_mode():
+        m, _ = load_neutone_model(model_file)
+        print(
+            f"Running benchmark for buffer sizes {buffer_size} and sample rates {sample_rate}"
+        )
+        for sr, bs in itertools.product(sample_rate, buffer_size):
+            m.set_daw_sample_rate_and_buffer_size(sr, bs)
+            m.reset()
+            for _ in range(number):  # Warmup
+                m.forward(torch.rand((2, bs)))
+            m.reset()
+            durations = timeit.repeat(
+                lambda: m.forward(torch.rand((2, bs))), repeat=repeat, number=number
```

Comment:
For a slightly more accurate benchmark, you could generate the random audio blocks before timing so the torch memory allocations aren't included. But being conservative probably isn't a bad thing here.

Reply:
I tried a version of that after sending the PR. I'm not that convinced, on my machine I only see a noticeable difference for clipper (~20-30%) because generally the model is much slower than creating the random numbers. The only downside is that it makes the code a bit more confusing, but I could refactor what I had a bit.

Reply:
Yeah I think it's fine to leave it as is

Reply:
I added this eventually